in_toto/verifylib.py

```diff
@@ -1137,64 +1105,47 @@ def verify_item_rules(source_name, source_type, rules, links):
-    # depending on the source_type
     if rule_type == "match":
-      source_artifacts_queue = verify_match_rule(
+      consumed = verify_match_rule(
           rule, source_artifacts_queue, source_artifacts, links)
 
     elif rule_type == "allow":
-      source_artifacts_queue = verify_allow_rule(rule, source_artifacts_queue)
-
-    elif rule_type == "disallow":
-      verify_disallow_rule(rule, source_artifacts_queue)
+      consumed = verify_allow_rule(rule, source_artifacts_queue)
 
-
-    # CREATE, DELETE and MODIFY always operate either on products, on materials
-    # or both, independently of the source_type ...
     elif rule_type == "create":
-      source_products_queue = verify_create_rule(
+      consumed = verify_create_rule(
           rule, source_materials_queue, source_products_queue)
 
-      # The create rule only updates the products_queue, which in turn
-      # only affects the generic artifacts queue if source_type is "products"
-      if source_type == "products":
-        source_artifacts_queue = source_products_queue
-
     elif rule_type == "delete":
-      source_materials_queue = verify_delete_rule(
+      consumed = verify_delete_rule(
           rule, source_materials_queue, source_products_queue)
 
-      # The delete rule only updates the materials_queue, which in turn
-      # only affects the generic artifacts queue if source_type is "materials"
-      if source_type == "materials":
-        source_artifacts_queue = source_materials_queue
-
-    # NOTE: Can't reach `else` branch, if the rule is none of these types
-    # an exception would have been raised above in `unpack_rule`
-    elif rule_type == "modify": # pragma: no branch
-      # The modify rule updates materials_queue and products_queue. We have to
-      # update the generic artifacts queue accordingly.
-      if source_type == "materials":
-        source_materials_queue, source_products_queue = verify_modify_rule(
-            rule, source_artifacts_queue, source_products_queue,
-            source_materials, source_products)
-        source_artifacts_queue = source_materials_queue
-
-      # NOTE: Can't reach `else` branch, if the source_type is none of these
-      # types an exception would have been raised above in `unpack_rule`
-      elif source_type == "products": # pragma: no branch
-        source_materials_queue, source_products_queue = verify_modify_rule(
-            rule, source_materials_queue, source_artifacts_queue,
-            source_materials, source_products)
-        source_artifacts_queue = source_products_queue
+    elif rule_type == "modify":
+      consumed = verify_modify_rule(
+          rule, source_materials_queue, source_artifacts_queue,
+          source_materials, source_products)
+
+    # Fail
+    elif rule_type == "disallow":
+      verify_disallow_rule(rule, source_artifacts_queue)
+
+    # Unreachable
+    else: # pragma: no cover
+      pass
```

Comment:
Would it be useful to log this in case it's really not unreachable? (I know, I know)

Reply:
I still don't know you to universally best handle these. I once triggered are related discussion but we never agreed on a best practice (see https://github.com/secure-systems-lab/code-style-guidelines/issues/18).

Reply:
But yes, pass is a bad idea. Maybe raise a custom `ErrorUnreachable`? :)